pkg/apis/openebs.io/scp/v1alpha1/storage-pool.go

```diff
@@ -29,8 +31,387 @@ import (
 type StoragePool struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec is the spec for a StoragePool resource
+	Spec StoragePoolSpec `json:"spec"`
+	// Status is for handling status of StoragePool resource
+	Status StoragePoolStatus `json:"status,omitempty"`
 }
 
+// StoragePoolSpec is the spec for a StoragePool resource
+type StoragePoolSpec struct {
+	// StorageCohortID of the cohort the pool is a part of
+	StorageCohortID string `json:"storageCohortID,omitempty"`
+
+	// Type of the storage pool, viz lvm, spdk blobstor
+	Type StoragePoolType `json:"type,omitempty"`
+
+	// PoolCofiguration based on the above type
+	PoolCofiguration StoragePoolCofiguration `json:"typeSpecificConfiguration,omitempty"`
+
+	// DeviceSpec for device related filtering, configuration
+	DeviceSpec StoragePoolDeviceSpec `json:"deviceSpec,omitempty"`
+
+	// Capabilities of the Storage pool and the supported configurations
+	Capabilities StoragePoolCapabilities `json:"capabilities,omitempty"`
+
+	// RequestedCapacity for the pool creation
+	RequestedCapacity resource.Quantity `json:"requestedCapacity,omitempty"`
+}
+
+// StoragePoolStatus is for handling status of StoragePool resource
+type StoragePoolStatus struct {
+	// ReferenceResource
+	ReferenceResource StoragePoolReferenceResource `json:"referenceResource,omitempty"`
```

Comment:
What is this reference?

Reply:
Not sure, created it as per the Design Document.